### PR TITLE
Semantic highlight intradoclinks in documentation

### PR DESCRIPTION
**Pull request by Veykril**
Changelog feature highlight Intradoclinks in doc comments

### CODE REVIEW

**Comment by Veykril** [comment]
Our intradoclink resolution certainly needs some improvement, most links I tried making didn't seem to work, here `noop` is a macro in the same file so it should be highlight but it isn't.
firefox_7djeQwRgTy

File: crates/rust-analyzer/src/semantic_tokens.rs [hunk 1]
```diff
@@ -45,15 +45,16 @@ define_semantic_token_types![
...
     (COLON, "colon"),
+    (COMMA, "comma"),
+    (CONST_PARAMETER, "constParameter"),
     (DOT, "dot"),
     (ESCAPE_SEQUENCE, "escapeSequence"),
     (FORMAT_SPECIFIER, "formatSpecifier"),
     (GENERIC, "generic"),
-    (CONST_PARAMETER, "constParameter"),
-    (LIFETIME, "lifetime"),
+    (INTRA_DOC_LINK, "intraDocLink"),
```
Comment (matklad) [inline crates/rust-analyzer/src/semantic_tokens.rs:55]: Do we need a separate token *type* here? I think that we might want to set the type to the type of the pointed to item, and set modifier to "doc link". That is, treat links (the part that is a rust path) as small snippets of injected rust code.

Comment (Veykril) [reply crates/rust-analyzer/src/semantic_tokens.rs:55]: Oh that's a nice idea actually

**Comment by lnicola** [comment]
image

**Pull request merged**